src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRun/DelayLoadHelperThunk.cs

```diff
@@ -0,0 +1,109 @@
+        {
+            _helperCell = factory.GetReadyToRunHelperCell(helperId);
+            _instanceCell = instanceCell;
+            _moduleImport = factory.ModuleImport;
+        }
+
+        public void AppendMangledName(NameMangler nameMangler, Utf8StringBuilder sb)
+        {
+            sb.Append("DelayLoadHelper->");
+            _instanceCell.AppendMangledName(nameMangler, sb);
+        }
+
+        protected override string GetName(NodeFactory factory)
+        {
+            return "DelayLoadHelper";
+        }
+
+        public override ObjectData GetData(NodeFactory factory, bool relocsOnly)
+        {
+            ObjectDataBuilder builder = new ObjectDataBuilder();
+            builder.AddSymbol(this);
+
+            switch (factory.Target.Architecture)
+            {
+                case TargetArchitecture.X64:
+                    {
+                        if (!relocsOnly)
+                        {
+                            builder.RequireInitialAlignment(8);
+
+                            // lea rax, [pCell]
+                            builder.EmitByte(0x48);
+                            builder.EmitByte(0x8D);
+                            builder.EmitByte(0x05);
+                        }
+                        builder.EmitReloc(_instanceCell, RelocType.IMAGE_REL_BASED_REL32);
+
+                        if (!relocsOnly)
+                        {
+                            // push table index
+                            builder.EmitByte(0x6A);
+                            builder.EmitByte((byte)_instanceCell.Table.IndexFromBeginningOfArray);
+
+                            // push [module]
+                            builder.EmitByte(0xFF);
+                            builder.EmitByte(0x35);
+                        }
+                        builder.EmitReloc(_moduleImport, RelocType.IMAGE_REL_BASED_REL32);
+
+                        if (!relocsOnly)
+                        {
+                            // TODO: additional tricks regarding UNIX AMD64 ABI
+
+                            // jmp [helper]
+                            builder.EmitByte(0xFF);
+                            builder.EmitByte(0x25);
+                        }
+                        builder.EmitReloc(_helperCell, RelocType.IMAGE_REL_BASED_REL32);
+
+                        break;
+                    }
+
+                default:
+                    throw new NotImplementedException();
+            }
+
+            return builder.ToObjectData();
+        }
+
+        public override ObjectNodeSection Section => ObjectNodeSection.TextSection;
+
+        int ISymbolDefinitionNode.Offset => 0;
+        int ISymbolNode.Offset => 0;
+        public override bool IsShareable => false;
+        public override bool StaticDependenciesAreComputed => true;
+        protected override int ClassCode => 433266948;
+    }
+
+    public class DelayLoadHelperThunk_Obj : DelayLoadHelperThunk
```

Comment:
Why do we need this?

Reply:
My assumption was that this would correspond to the READYTORUN_HELPER_DelayLoad_Helper_Obj ID but perhaps I just don't yet sufficiently understand the logic for choosing the ID's - it seems to me that the ID's are basically per helper type.